ggshield/core/url_utils.py

```diff
@@ -24,24 +24,32 @@ def clean_url(url: str, warn: bool = False) -> ParseResult:
     return parsed_url
 
 
-def dashboard_to_api_url(dashboard_url: str, warn: bool = False) -> str:
+def validate_instance_url(url: str, warn: bool = False):
     """
-    Convert a dashboard URL to an API URL.
-    handles the SaaS edge case where the host changes instead of the path
+    Validate a dashboard URL
     """
-    parsed_url = clean_url(dashboard_url, warn=warn)
+    parsed_url = clean_url(url, warn=warn)
     if parsed_url.scheme != "https" and not (
         parsed_url.netloc.startswith("localhost")
         or parsed_url.netloc.startswith("127.0.0.1")
     ):
-        raise UsageError(
-            f"Invalid scheme for dashboard URL '{dashboard_url}', expected HTTPS"
-        )
+        raise UsageError(f"Invalid scheme for dashboard URL '{url}', expected HTTPS")
     if any(parsed_url.netloc.endswith("." + domain) for domain in GITGUARDIAN_DOMAINS):
         if parsed_url.path:
             raise UsageError(
-                f"Invalid dashboard URL '{dashboard_url}', got an unexpected path '{parsed_url.path}'"
+                f"Invalid dashboard URL '{url}', got an unexpected path '{parsed_url.path}'"
             )
+
+
+def dashboard_to_api_url(dashboard_url: str, warn: bool = False) -> str:
+    """
+    Convert a dashboard URL to an API URL.
+    handles the SaaS edge case where the host changes instead of the path
+    """
+    validate_instance_url(dashboard_url, warn=warn)
+
+    parsed_url = clean_url(dashboard_url, warn=warn)
```

Comment:
You already do this in `validate_instance_url` is there a reason to make a second call ?